File: src/db/models/Score.ts
// src/db/models/Score.ts
import { pgTable, varchar, integer, uniqueIndex } from "drizzle-orm/pg-core";

export const Score = pgTable("scores", {
  discordID: varchar("discordID").notNull(), // Ensure discordID is not null
  roundID: varchar("roundID").notNull(), // Ensure roundID is not null
  score: integer("score").notNull(), // Ensure score is not null
  tagNumber: integer("tagNumber"), // Optional tag field
});

// Define a unique index for the combination of discordID and roundID
export const ScoreUniqueIndex = uniqueIndex("unique_round").on(
  Score.discordID,
  Score.roundID
);

File: db/bundb/migrations/20241203151046_init/score.go
package models

import "github.com/uptrace/bun"

// Score represents a score with DiscordID, RoundID, Score, and TagNumber.
type Score struct {
	bun.BaseModel `bun:"table:users,alias:u"`

	DiscordID string `bun:"discord_id,notnull"`
	RoundID   string `bun:"round_id,notnull"`
	Score     int    `bun:"score,notnull"`
	TagNumber int    `bun:"tag_number"`
}
